base/download.jl

```diff
@@ -19,29 +19,48 @@ if Sys.iswindows()
             end
             pipeline_error(proc)
         end
-        filename
+        return filename
     end
-else
-    function download(url::AbstractString, filename::AbstractString)
-        if Sys.isapple() && Sys.isexecutable("/usr/bin/curl")
-            run(`/usr/bin/curl -g -L -f -o $filename $url`) # issue #30956
-        elseif Sys.which("curl") !== nothing
-            run(`curl -g -L -f -o $filename $url`)
-        elseif Sys.which("wget") !== nothing
-            try
-                run(`wget -O $filename $url`)
-            catch
-                rm(filename, force=true)  # wget always creates a file
-                rethrow()
-            end
-        elseif Sys.which("fetch") !== nothing
-            run(`fetch -f $filename $url`)
-        else
-            error("no download agent available; install curl, wget, or fetch")
+end
+
+function find_curl()
+    if Sys.isapple() && Sys.isexecutable("/usr/bin/curl")
+        "/usr/bin/curl"
+    elseif Sys.iswindows() && Sys.isexecutable(joinpath(get(ENV, "SYSTEMROOT", "C:\\Windows"), "System32\\curl.exe"))
```

Comment:
Is it possible to just look for "curl.exe"? That way if it is on the path elsewhere it will work too.

Reply:
Not really, because applications and other programs can modify the path to set a higher precedence.  E.g. anaconda screws it up the precedence by making its shipped curl to overtake the system provided version. See also https://github.com/JuliaLang/julia/pull/30969